deploy/ansible/roles/iota-core-node/templates/docker-compose-iota-core.yml.j2

```diff
@@ -9,58 +9,120 @@
+      - "8080:8080/tcp"   # REST-API
+      - "8081:8081/tcp"   # Dashboard
+      - "9311:9311/tcp"   # Prometheus
+      - "9089:9029/tcp"   # INX
     volumes:
-      - ./snapshot.bin:/app/data/snapshot.bin:ro
       - ./config.json:/app/config.json:ro
       - ./data:/app/data/
+      - ./snapshot.bin:/app/data/snapshot.bin:ro
       - /etc/localtime:/etc/localtime:ro
-    ports:
-      - "0.0.0.0:14666:14666/tcp"
-      - "0.0.0.0:8080:8080/tcp"
-      - "0.0.0.0:8081:8081/tcp"
-      - "0.0.0.0:6061:6061/tcp"
-      # prometheus
-      - "0.0.0.0:9311:9311/tcp"
-    environment:
-      - WEBAPI_BINDADDRESS=0.0.0.0:8080
-      - DASHBOARD_BINDADDRESS=0.0.0.0:8081
-      - PROFILING_BINDADDRESS=0.0.0.0:6061
-    command: >
-      -c config.json
-      --logger.level=debug
-      --logger.disableCaller=false
-      --logger.disableStacktrace=false
-      --logger.encoding=console
-      --logger.outputPaths=stdout
-      --database.path=/app/data/database
-      --p2p.db.path=/app/data/peerdb
-      --profiling.bindAddress=0.0.0.0:6061
-      --profiling.enabled=true
-      --protocol.snapshot.path=/app/data/snapshot.bin
+    command:
+      - "-c"
+      - "config.json"
+      - "--logger.level=debug"
+      - "--logger.disableCaller=false"
+      - "--p2p.peers=/dns/node-01.feature/tcp/14666/p2p/12D3KooWCrjmh4dUCWfGVQT6ivzArieJB9Z3eKdy2mdEEN95NDPS"
+      - "--p2p.externalMultiAddresses={{ ips | join(',') }}"
+      - "--p2p.identityPrivateKey={{p2pIdentityPrvKey}}"
+      - "--p2p.db.path=/app/data/peerdb"
+      - "--profiling.enabled=true"
+      - "--profiling.bindAddress=0.0.0.0:6061"
+      - "--restAPI.bindAddress=0.0.0.0:8080"
+      - "--database.path=/app/data/database"
+      - "--protocol.snapshot.path=/app/data/snapshot.bin"
       {% if 'node-01' in inventory_hostname or 'node-02' in inventory_hostname or 'node-03' in inventory_hostname %}
-      --validator.enabled=true
-      --validator.account={{validatorAccount}}
-      --validator.privateKey={{validatorPrivKey}}
+      - "--validator.enabled=true"
+        {% if 'node-01' in inventory_hostname %}
+      - "--validator.ignoreBootstrapped=true"
+        {% endif %}
+      - "--validator.account={{validatorAccount}}"
+      - "--validator.privateKey={{validatorPrvKey}}"
       {% endif %}
-      {% if 'node-01' in inventory_hostname %}
-      --validator.ignoreBootstrapped=true
-      {% endif %}
-      --p2p.peers=/dns/node-01.feature/tcp/14666/p2p/12D3KooWCrjmh4dUCWfGVQT6ivzArieJB9Z3eKdy2mdEEN95NDPS
-      --p2p.externalMultiAddresses={{ ips | join(',') }}
-      --p2p.identityPrivateKey={{p2pIdentityPrivateKey}}
-      --inx.enabled=true
-      --inx.bindAddress=iota-core:9029
+      - "--dashboard.bindAddress=0.0.0.0:8081"
+      - "--metrics.bindAddress=iota-core:9311"
+      - "--inx.enabled=true"
+      - "--inx.bindAddress=iota-core:9029"
+
+##################
+# INX Extensions #
+##################
 
+{% if 'node-04' in inventory_hostname %}
```

Comment:
The indexer is actually needed on every host if you want to spam: the evil spammer uses the indexer to obtain the outputs it can spend.

Reply:
good point